src/web/screens/sharingPermissions.tsx

```diff
@@ -91,15 +125,24 @@ function SharingPermissions() {
     }
   };
 
-  const loadSharingList = useCallback(async () => {
-    const response = await GetSharingList();
-    setSharedSiteIds(response.allowed_sites);
-    setSharedTypes(response.allowed_types ?? []);
-  }, []);
-
   useEffect(() => {
+    const loadSharingList = async () => {
+      try {
+        const response = await GetSharingList();
+        setSharedSiteIds(response.allowed_sites);
+        setSharedTypes(response.allowed_types ?? []);
+      } catch (e: unknown) {
+        if (e instanceof ApiError) {
+          if (e.statusCode === 404) {
+            setShowErrorDialog(true);
+            return;
+          }
+          throwError(e);
```

Comment:
if there is any other ApiError for this get request, I think we should still show error page